modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/counter/BaseAMImageCounter.java

```diff
@@ -50,46 +62,104 @@ public int countExpectedAMImageEntries(long companyId) {
 		return counter.get();
 	}
 
-	protected abstract void forEachFileEntry(
-			long companyId, Consumer<DLFileEntry> consumer)
-		throws PortalException;
-
-	protected String[] getMimeTypes() {
-		return ArrayUtil.filter(
-			amImageMimeTypeProvider.getSupportedMimeTypes(),
-			amImageValidator::isProcessingSupported);
-	}
+	protected abstract String getClassName();
 
 	@Reference
 	protected AMImageMimeTypeProvider amImageMimeTypeProvider;
 
 	@Reference
 	protected AMImageValidator amImageValidator;
 
+	@Reference
+	protected ClassNameLocalService classNameLocalService;
+
 	@Reference
 	protected DLFileEntryConfigurationProvider dlFileEntryConfigurationProvider;
 
-	private Consumer<DLFileEntry> _getCountDLFileEntryConsumer(
-		AtomicInteger counter) {
+	@Reference
+	protected DLFileEntryLocalService dlFileEntryLocalService;
+
+	private void _forEachGroup(long companyId, Consumer<Group> groupConsumer)
+		throws PortalException {
+
+		ActionableDynamicQuery actionableDynamicQuery =
+			GroupLocalServiceUtil.getActionableDynamicQuery();
+
+		actionableDynamicQuery.setCompanyId(companyId);
+		actionableDynamicQuery.setModelClass(Group.class);
+		actionableDynamicQuery.setPerformActionMethod(
+			(ActionableDynamicQuery.PerformActionMethod<Group>)
+				groupConsumer::accept);
+
+		actionableDynamicQuery.performActions();
+	}
+
+	private Consumer<Group> _getCountDLFileEntryConsumer(
+		long companyId, AtomicInteger counter,
+		long previewableProcessorMaxSize) {
 
 		Map<Long, Long> groupPreviewableProcessorMaxSizeMap =
 			dlFileEntryConfigurationProvider.
 				getGroupPreviewableProcessorMaxSizeMap();
 
-		return dlFileEntry -> {
-			Long previewableProcessorMaxSize =
-				groupPreviewableProcessorMaxSizeMap.get(
-					dlFileEntry.getGroupId());
+		return group -> {
+			long groupId = group.getGroupId();
+
+			Long previewableGroupProcessorMaxSize =
+				groupPreviewableProcessorMaxSizeMap.getOrDefault(
+					groupId, previewableProcessorMaxSize);
```

Comment:
Sorry, I didn't see this before: don't use wrapper types unless strictly necessary.